c10/xpu/XPUCachingAllocator.cpp

```diff
@@ -728,27 +747,24 @@ class DeviceCachingAllocator {
       TORCH_CHECK(
           !active_pool,
           "torch.xpu.MemPool doesn't currently support expandable_segments.");
-      p.block =
-          try_allocate_expandable_block(device, p.queue(), p.pool, p.size());
+      p.block = try_allocate_expandable_block(
+          p.device(), p.queue(), p.pool, p.size());
       if (p.block && p.pool->owner_PrivatePool) {
         // The block is used only for XPU graph's PrivatePool.
         p.pool->owner_PrivatePool->allocation_count++;
       }
       return bool(p.block);
-    }
-    void* ptr = sycl::aligned_alloc_device(
-        kDeviceAlignment,
-        size,
-        xpu::get_raw_device(device),
-        xpu::get_device_context());
-    if (!ptr) {
-      return false;
+    } else {
+      allocPrimitive(&ptr, size, p);
```

Comment:
Here the logic is different from CUDA.
In CUDA, they will check whether we are under graph capturing mode.
Why do we not need it?
@majing921201  pls. help co-review.

Reply:
@guangyey , we need to check with the SYCL runtime team whether `sycl::aligned_alloc_device` is guaranteed to support graph mode.

Reply:
LGTM, XPU graph test is pass with `sycl::aligned_alloc_device`